core/trino-main/src/main/java/io/trino/server/security/oauth2/NimbusClient.java

```diff
@@ -78,54 +93,61 @@
 {
     private static final Logger LOG = Logger.get(NimbusAirliftHttpClient.class);
 
+    private final boolean enableDiscovery;
+    private final Duration discoveryTimeout;
+    private final boolean userInfoEndpointEnabled;
     private final Issuer issuer;
     private final ClientID clientId;
     private final ClientSecretBasic clientAuth;
     private final Scope scope;
-    private final URI authUrl;
-    private final URI tokenUrl;
-    private final Optional<URI> userinfoUri;
+    private final String principalField;
+    private final Set<String> accessTokenAudiences;
     private final NimbusHttpClient httpClient;
-    private final JWSKeySelector<SecurityContext> jwsKeySelector;
-    private final JWTProcessor<SecurityContext> accessTokenProcessor;
     private final AuthorizationCodeFlow flow;
+    private URI authUrl;
+    private URI tokenUrl;
+    private URI jwksUrl;
+    private Optional<URI> userinfoUrl;
+    private Optional<Issuer> accessTokenIssuer;
+    private JWSKeySelector<SecurityContext> jwsKeySelector;
+    private JWTProcessor<SecurityContext> accessTokenProcessor;
```

Comment:
Can we extract a configuration as separate component, where you could track how they are overwritten?